Daily summary text

alexander0042 opened the pull request:
## Describe the change

## Type of change
Extend the initial work on daily text summaries to each day, building off the wonderful work by @cloneofghosts!

- [ ] Bugfixes to existing code
- [ ] Breaking change
- [ ] New API Version
- [X] General Improvement
- [ ] Code quality improvements to existing code or addition of tests
- [ ] Documentation Updates

## Checklist

- This pull request fixes issue: fixes #
- [ ] Code builds locally. **Your pull request won't be merged unless tests pass**
- [ ] Code has been formatted using ruff

cloneofghosts commented:
I noticed a small issue with the daily summary generation and am working on fixing. Do you want me to commit here or create a separate branch?

cloneofghosts commented:
> * It seems like `calculate_day_text` really prefers 24 hour periods, since I seemed to be issues with the length of `periodStats` otherwise? As a first attempt, I expanded the processing to run 197 hours (8 days + 5 hours), to cover the full range, but I'm wondering if this is a bug or what we expect?  It's not a showstopper, but daylight savings time means that there are 23/25 hour days, although not really often enough to worry about so long as it's close enough.

The function should allow you to send less than 24 hours of data to it and should still work. Forgot about daylight savings and it should probably be updated to 25. Was the function crashing with less than 24 hours or was something else happening?

> * I'm going to think through the way we handle Clear/ Mostly clear conditions, since at the moment, it can give sentences like "Clear starting in the evening", which I don't know if that makes sense? For any other condition, absolutely (and I like the logic here), but since clear is something of a default, it ended up sounding a little flunky.

The way the function works currently is it calculates the average cloud cover for the period and then it uses whatever the most common cloud cover is as the summary or if there's a tie then whichever one comes first is used. If you have a better idea on how to calculate it then feel free to update the logic.

cloneofghosts commented:
> My plan is to get this stable/ working, then push it all out as a V2.7 release! These text summaries don't have to be perfect, just better than we currently do (which I think they already are)! It'll also use the new ingest files, so will be good to have everything on the same footing

Missed your comment when I added mine. So does this mean that the items marked as 2.6.1 and 2.6.2 [here](https://github.com/Pirate-Weather/pirateweather/discussions/414) are being moved to 2.7 or a 2.7.x release? Either way is fine for me just curious as to what the plan is.

alexander0042 commented:
Pretty much! I updated that roadmap discussion, since the overall trend is still the same, just moved some things around to better reflect where progress has been made

cloneofghosts commented:
> Pretty much! I updated that roadmap discussion, since the overall trend is still the same, just moved some things around to better reflect where progress has been made

Alright, makes sense. Anything that isn't on that list I assume is something that doesn't have a timeline on when it would be added? No big deal just curious is all.

I'll commit my changes later tonight so I don't mess up anything you're working on.

alexander0042 commented:
Anything not on that list is for the future... unless it looks interesting or the code appears (which is why daily text summaries are in there now!), then there's always some wiggle room

cloneofghosts commented:
@alexander0042 Did you move this up to dev for testing? When I visit the dev endpoint I see the new summaries.

Doing a quick look at what is there I see a few issues:

1. The last day doesn't have a summary. It just says Clear.
2. One of the summaries is showing this: Partly cloudy until night, starting again in the afternoon. Which reading it doesn't make sense for a daily summary. I also see this Drizzle in the afternoon. for a day in which no precipitation is forecasted. 

For item 2 I wonder if the daily start/end points are slightly off and it's leading to the weird summaries? The summary for the current day is correct but after that its a bit off.

alexander0042 commented:
I've structured it so that the `hourly.summary` should give the next 24 hours (corresponding to the data); however, the `daily[0-7]summary` gives the 24 hour daily summaries, from 4am-4am. That's what required the other changes to the hourly interpolation, since now objects are generated from 00:00 on day 0 (request day) to 04:00 on day 8. I think it's more or less aligns with what people might expect it to do, which is as good of logic as any

cloneofghosts commented:
I was mainly basing my feedback on whichever version of the daily summaries is available on dev at the moment. Not sure if it's up-to-date with this PR or is running an outdated version.

Mainly because I see things like:

* Mostly cloudy until night, starting again in the afternoon.
* Partly cloudy in the evening and morning.
* Mostly clear until morning.
* Light rain until morning.
* Light rain until morning.

Those summaries makes sense if it were the next 24 hour summaries but for a day summary they don't. Again no idea if dev is running the same code as this PR.

cloneofghosts commented:
@alexander0042 Since the day summaries are already on dev and if this fixes the issues with the weird summaries then maybe we should merge this in and create additional PRs to fix any bugs?

That way I can see how the latest version is generating summaries and give proper feedback.

cloneofghosts commented:
Not sure if this is related to the new summaries but am getting Internal Server Error when testing with Tokyo, JP (35.6895,139.69171). @alexander0042 

Other locations with just GFS/GEFS data work so maybe something funky going on there?

alexander0042 commented:
Yea, that's a really weird error. I missed a try catch block, so that's why it's throwing a 500, but I don't see the error in my local env??? From the logs I can see it is an issue with the text generation, but until I can replicate it in testing (which should be the same), it's hard to get to root

alexander0042 commented:
Edit: it's actually an issue with translations, so that might explain the difference
